prettyPrinters/providers.py

```diff
@@ -124,8 +126,10 @@ def StdStringSummaryProvider(valobj, dict):
 
     vec = valobj.GetChildAtIndex(0)
     length = vec.GetNumChildren()
-    chars = [chr(vec.GetChildAtIndex(i).GetValueAsUnsigned()) for i in range(length)]
-    return '"%s"' % "".join(chars)
+    chars = [vec.GetChildAtIndex(i).GetValueAsUnsigned() for i in range(length)]
+    if system() == "Darwin":
+        return '"%s"' % "".join(chr(char) for char in chars)
+    return '"%s"' % b''.join(bytes(bytearray([char])) for char in chars).decode(encoding='UTF-8')
```

Comment:
Not sure whether it's possible. Unlike `&str`, `String` contains a vector with overridden children (because of pretty-printer for `Vec`), so it's not possible to get something like `data_ptr` here.

Reply:
Also, @mchernyavsky could you please look at this code? Do you think there is a straightforward way for Python 3?

Reply:
Can be simplified:

```suggestion
    return '"%s"' % bytes(chars).decode(encoding='UTF-8')
```